transportmiddleware/transportmiddleware.go

```diff
@@ -0,0 +1,33 @@
+package transportmiddleware
+
+import (
+	"context"
+
+	"github.com/yarpc/yab/transport"
+)
+
+var registeredMiddleware TransportMiddleware
+
+// Register sets the provided transport middleware to be used on future
+// calls to Apply(). Calls to Register() will overwrite previously registered
+// middlewares; that is, only one middleware is allowed at a time.
+func Register(tm TransportMiddleware) {
+	registeredMiddleware = tm
```

Comment:
We should consider thread-safety here. Let's acquire a write lock before changing the value.